base/tuple.jl

```diff
@@ -142,6 +146,7 @@ ntuple(f, ::Type{Val{15}}) = (@_inline_meta; (f(1), f(2), f(3), f(4), f(5), f(6)
 
 function ntuple(f::F, ::Type{Val{N}}) where {F,N}
     Core.typeassert(N, Int)
+    (N >= 0) || throw(ArgumentError(string("tuple length should be ≥0, got ", N)))
```

Comment:
Sure. Actually, for `0<=N<=15` there's explicit `ntuple(identity, Val{N})`, the check for `N<0` is in a generic method.
In fact, `ntuple(f, Val{N})` is substantially slower for `N>=16`. On the 11 days old master (b838f2eec6)
```
julia> @benchmark ntuple(identity, Val{15})
BenchmarkTools.Trial: 
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     3.365 ns (0.00% GC)
  median time:      3.379 ns (0.00% GC)
  mean time:        3.400 ns (0.00% GC)
  maximum time:     16.001 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     1000

julia> @benchmark ntuple(identity, Val{16})
BenchmarkTools.Trial: 
  memory estimate:  288 bytes
  allocs estimate:  2
  --------------
  minimum time:     418.864 ns (0.00% GC)
  median time:      423.166 ns (0.00% GC)
  mean time:        448.653 ns (2.32% GC)
  maximum time:     6.745 μs (87.57% GC)
  --------------
  samples:          10000
  evals/sample:     199
```
Few nanoseconds could be explained by `Base.@_inline_meta` absence, but not hundreds.